src/systemtask/SystemTask.cpp

```diff
@@ -200,6 +203,20 @@ void SystemTask::Work() {
   while (true) {
     UpdateMotion();
 
+    if (activityController.UpdateSteps(motionController.NbSteps()) &&
+        settingsController.GetActivity() == Controllers::Settings::Activity::On) {
+      NRF_LOG_INFO("activity");
+      Controllers::NotificationManager::Notification notif;
+      constexpr char message[] = "Low activity\0Fewer steps than threshold in last hour. Try walking around.";
+      constexpr uint8_t messageSize =
+        std::min(sizeof message / sizeof(char), Pinetime::Controllers::NotificationManager::MaximumMessageSize());
+      std::memcpy(notif.message.data(), message, messageSize * sizeof(char));
+      notif.size = messageSize;
+      notif.category = Pinetime::Controllers::NotificationManager::Categories::SimpleAlert;
+      notificationManager.Push(std::move(notif));
+      PushMessage(Messages::OnNewNotification);
+    }
+
```

Comment:
This should be moved to at least one function. Also I'm wondering if this should be part of DisplayApp. What DisplayApp and SystemTask should do is not defined at all, but considering that DisplayAppRecovery does not benefit from this, it probably should not be part of the "system", but rather the "application". I'm working on fixing the ambiguity.

Reply:
I agree that this should be moved to a function, but as you say, I'm not sure whether that function should be in NotificationManager, DisplayApp or even SystemTask. Because of this I won't move this to a function yet.

#1482 makes constructing a notification a lot simpler as well, so when that gets cleaned up and merged it might not even need to be moved to its own function.